### PR TITLE
BugzID: 12534 respect reader lists.

### PR DESCRIPTION
**Pull request by rnewson**

### CODE REVIEW

File: src/chttpd_db.erl [hunk 1]
```diff
@@ -194,6 +194,8 @@ delete_db_req(#httpd{}=Req, DbName) ->
     end.
 
 do_db_req(#httpd{path_parts=[DbName|_], user_ctx=Ctx}=Req, Fun) ->
+    SecObj = fabric:get_security(DbName),
+    check_is_reader(Ctx, SecObj),
```
Comment (kocolosk) [inline src/chttpd_db.erl]: Perhaps it's good that we're explicit here, but I think the call to `check_is_reader` is redundant.  `get_security/1` opens the database shard using `couch_db:open/2` and thus triggers a reader list check.

Comment (rnewson) [inline src/chttpd_db.erl]: fabric:get_security/1 does call check_is_reader but does so as admin, so we need to explicit check.

Comment (kocolosk) [inline src/chttpd_db.erl]: Yes, quite right.  `fabric:get_security/2` with the user's context should do the check, though.

**Comment by kocolosk** [comment]
Looks good @rnewson.  Can you squash the commits and adhere to the (just posted) [commit message guidelines](https://github.com/cloudant/bigcouch/wiki/Writing-Commit-Messages)?

**Pull request merged**